### PR TITLE
SDXL: Add Cache Dataset

### PR DESCRIPTION
**Pull request by zhanghuiyao**

### CODE REVIEW

File: examples/stable_diffusion_xl/gm/data/dataset_cache.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import numpy as np
+import pandas as pd
+
+
+class Text2ImageCacheDataset:
+    def __init__(self, data_path, cache_path):
+        super().__init__()
+        self.dataset_column_names = ["latent", "vector", "crossattn"]
+        self.dataset_output_column_names = ["latent", "vector", "crossattn"]
```
Comment (hadipash) [inline examples/stable_diffusion_xl/gm/data/dataset_cache.py:11]: Why are two column name fields needed?

File: examples/stable_diffusion_xl/gm/data/dataset_cache.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return batch_latent, batch_vector, batch_crossattn
+
+    def __len__(self):
+        return len(self.all_files)
+
+    @staticmethod
+    def list_cache_files_recursively(data_path):
+        anno_dir = data_path
+        anno_list = sorted(
+            [os.path.join(anno_dir, f) for f in list(filter(lambda x: x.endswith(".csv"), os.listdir(anno_dir)))]
```
Comment (hadipash) [inline examples/stable_diffusion_xl/gm/data/dataset_cache.py:45]: `Path(anno_dir).glob('*.csv')`

**Review by hadipash — COMMENTED** [review]
By how much does the training speed improve when using cached latents?

**Comment by zhanghuiyao** [comment]
> By how much does the training speed improve when using cached latents?

~450ms

File: examples/stable_diffusion_xl/vanilla_finetune.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+### 4. Support List
+
+<div align="center">
+
+| Model Name | Device      | Cards  | MindSpore        | CANN | ImageSize | Compile Cost | PerBatchSize | FlashAttention | Cache | DataSink | Step Time |
+|------------|-------------|--------|------------------|------|-----------|--------------|--------------|----------------|-------|----------|-----------|
+| SDXL-Base  | Ascend 910* | 1*8p   | mindspore 2.2.10 | C15  | 1024x1024 | 30min        | 1            | False          | False | False    | 1.10s     |
+| SDXL-Base  | Ascend 910* | 1*8p   | mindspore 2.2.10 | C15  | 1024x1024 | 30min        | 1            | True           | True  | True     | 0.74s     |
+| SDXL-Base  | Ascend 910* | 1*8p   | mindspore 2.2.10 | C15  | 1024x1024 | 30min        | 6            | True           | False | False    | 2.70s     |
+| SDXL-Base  | Ascend 910* | 1*8p   | mindspore 2.2.10 | C15  | 1024x1024 | 30min        | 6            | True           | True  | True     | 2.15s     |
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/vanilla_finetune.md:94]: Are the compilation time really the same for all settings?

Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/vanilla_finetune.md:94]: > Are the compilation time really the same for all settings?

Currently, it seems to be quite similar

**Review by geniuspatrick — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**